yaml/shared/src/main/scala/org/virtuslab/yaml/Position.scala

```diff
@@ -1,11 +1,24 @@
 package org.virtuslab.yaml
 
-final case class Position(offset: Int, line: Int, column: Int, input: Vector[String]):
+final case class Position(
+    offset: Int,
+    line: Int,
+    column: Int,
+    input: Vector[String],
+    endLine: Option[Int] = None,
+    endColumn: Option[Int] = None
```

Comment:
Hmm, I've got a problem with adding `endline` and `endColumn`. When I think about it, now `Position` becomes effectively a `Range` with an optional end Position parameter. Thus, maybe following 
```scala
final case class Position(
    offset: Int,
    line: Int,
    column: Int,
)

final case class Range(
    input: Vector[String],
    start: Position,
    end: Option[Position]
)
```
would be a better model for this?